src/backend/distributed/operations/worker_split_copy.c

```diff
@@ -0,0 +1,223 @@
+/*-------------------------------------------------------------------------
+ *
+ * worker_split_copy.c
+ *	 API implementation for worker shard split copy.
+ *
+ * Copyright (c) Citus Data, Inc.
+ *
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include <unistd.h>
+#include "c.h"
+#include "postgres.h"
+#include "catalog/namespace.h"
+#include "utils/lsyscache.h"
+#include "utils/builtins.h"
+#include "distributed/listutils.h"
+#include "distributed/metadata_cache.h"
+#include "distributed/relation_utils.h"
+#include "distributed/worker_split_copy.h"
+#include "distributed/worker_shard_copy.h"
+#include "distributed/relay_utility.h"
+
+typedef struct SplitCopyDestReceiver
+{
+	/* public DestReceiver interface */
+	DestReceiver pub;
+
+	/* Underlying shard copy dest receivers */
+	DestReceiver **shardCopyDestReceiverArray;
+
+	/* Split Copy Info */
+	SplitCopyInfo **splitCopyInfoArray;
+
+	/* Split factor */
+	uint splitFactor;
+
+	/* EState for per-tuple memory allocation */
+	EState *executorState;
+
+	/* Source shard Oid */
+	Oid sourceShardRelationOid;
+} SplitCopyDestReceiver;
+
+
+static void SplitCopyDestReceiverStartup(DestReceiver *dest, int operation,
+										 TupleDesc inputTupleDescriptor);
+static bool SplitCopyDestReceiverReceive(TupleTableSlot *slot,
+										 DestReceiver *dest);
+static void SplitCopyDestReceiverShutdown(DestReceiver *dest);
+static void SplitCopyDestReceiverDestroy(DestReceiver *copyDest);
+
+/*
+ * CreateSplitCopyDestReceiver creates a DestReceiver that performs
+ * split copy for sourceShardIdToCopy to destination split children
+ * based on splitCopyInfoList.
+ */
+DestReceiver *
+CreateSplitCopyDestReceiver(EState *executorState, uint64 sourceShardIdToCopy,
```

Comment:
Would it be possible to reuse PartitionedResultDestReceiver?

Reply:
Fixed.